brightener.cpp

```diff
@@ -1,32 +1,32 @@
 #include "brightener.h"
-
-ImageBrightener::ImageBrightener(Image& inputImage) : m_inputImage(inputImage) {
+ImageBrightener::ImageBrightener(unique_ptr<Image>& inputImage) {
+	m_inputImage = move(inputImage);
 }
 
 int ImageBrightener::BrightenWholeImage() {
 	// For brightening, we add a certain grayscale (25) to every pixel.
 	// While brightening, some pixels may cross the max brightness. They are
 	// called 'attenuated' pixels
 	int attenuatedPixelCount = 0;
-	for (int x = 0; x < m_inputImage.rows; x++) {
-		for (int y = 0; y < m_inputImage.columns; y++) {
-			if (m_inputImage.pixels[x * m_inputImage.columns + y] > (255 - 25)) {
+	for (int x = 0; x < m_inputImage->rows; x++) {
+		for (int y = 0; y < m_inputImage->columns; y++) {
+			if (m_inputImage->pixels[x * m_inputImage->columns + y] > (255 - 25)) {
 				++attenuatedPixelCount;
 				// m_inputImage.pixels[x * m_inputImage.rows + 1 + y] = 255;
-				m_inputImage.pixels[x * m_inputImage.columns + y] = 255;
+				m_inputImage->pixels[x * m_inputImage->columns + y] = 255;
```

Comment:
Try calculating the pixel offset once, instead of repeating the `x * m_inputImage.columns + y` calculation.
If you place this calculation in the `Image` struct, then the pixel-layout-logic gets encapsulated in it as well.